pghistory/core.py

```diff
@@ -237,10 +237,12 @@ def _generate_history_field(tracked_model, field):
     """
     field = tracked_model._meta.get_field(field)
 
+    db_column = field.db_column if field.db_column != field.name else None
```

Comment:
I'm trying to understand what this line is doing. Can you elaborate?

Reply:
Sure, it seems that after you initialize a field, Django always sets db_column to the name if it wasn't set by the user, even if it was set to None as it would be by default. This would result in db_column always being set to a non-null string. This would cause a problem for generating the AutoField because it would then always be explicitly setting the db_column on it, causing db_column to be set in the migrations even if the original Model didn't, and regenerating any migrations with AutoFields. 

So this checks if the field's db_column matches it's name, if they don't match the user must have set it, and if they do, the user didn't and it should be set to None.

Reply:
ah makes sense, thanks for clarifying!